### PR TITLE
fix(renumber): add prefix to NFA.renumber

### PR DESCRIPTION
**Pull request by Buitragox**
- Add a prefix string parameter to the renumber method

- Refactor the code using set and dict comprehension

- Simplify .gitignore

### CODE REVIEW

File: automathon/finite_automata/nfa.py [hunk 1]
```diff
@@ -377,13 +377,11 @@ def __get_e_closure(
             for st in self.delta[q][""]:
                 if st not in visited:
                     visited.append(st)
-                    ans.extend(
-                        [
-                            k
-                            for k in self.__get_e_closure(st, visited)
-                            if k not in ans
-                        ]
-                    )
```
Comment (Buitragox) [inline automathon/finite_automata/nfa.py:386]: I think the ruff formatter made some unintended changes, let me know if you want me to leave it like it was.

File: automathon/finite_automata/nfa.py [hunk 2]
```diff
@@ -522,39 +520,33 @@ def minimize(self) -> "NFA":
         local_nfa.renumber()
         return local_nfa
 
-    def renumber(self) -> None:
-        """Change the name of the states, renumbering each of the labels"""
-        idx = 0
-        new_tags = dict()
+    def renumber(self, prefix="") -> None:
```
Comment (rohaquinlop) [inline automathon/finite_automata/nfa.py:523]: `prefix` could be `q` by default

**Review by rohaquinlop — CHANGES_REQUESTED** [review]

File: automathon/finite_automata/nfa.py (hunk 2 above)
Comment (Buitragox) [reply automathon/finite_automata/nfa.py]: Done 👍

**Review by rohaquinlop — APPROVED** [review]
LGTM

**Pull request merged**